src/lib/dialog/dialog-ref.ts

```diff
@@ -42,9 +44,13 @@ export class MatDialogRef<T, R = any> {
   /** Result to be passed to afterClosed. */
   private _result: R | undefined;
 
+  /** Subscription to changes in the user's location. */
+  private _locationChanges: ISubscription = Subscription.EMPTY;
```

Comment:
Why is this `ISubscription` instead of `Subscription`?

Reply:
It's because the Angular `Location` isn't actually an observable and it doesn't return a regular `Subscription`.